spinalcordtoolbox/scripts/sct_detect_compression.py

```diff
@@ -0,0 +1,348 @@
+    """
+    Compute torsion as the average of absolute differences in orientation between the given slice and 3 slices
+    above and below. For details see Equation 3 and Supplementary Material in https://pubmed.ncbi.nlm.nih.gov/35371944/.
+    NOTE: As the torsion is computed from 3 slices above and below, it cannot be computed for the 3 first and last
+    3 slices --> `torsion = None` is used for the 3 first and 3 last slices.
+    :param metrics_agg_merged: dictionary with aggregated metrics
+    :param verbose: verbosity
+    :return: dictionary with torsion added
+    """
+    slices = list(metrics_agg_merged.keys())[3:-3]
+
+    for key in metrics_agg_merged.keys():  # Loop across slices
+        if key in slices:
+            try:
+                # Note: the key is a tuple (e.g. `1,`), not an int (e.g., 1), thus key[0] is used to convert tuple to int
+                # and `,` is used to convert int back to tuple
+                metrics_agg_merged[key]['Torsion'] = 1/6 * (abs(metrics_agg_merged[key]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] - 1,]['MEAN(orientation)']) +
+                                                            abs(metrics_agg_merged[key]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] + 1,]['MEAN(orientation)']) +
+                                                            abs(metrics_agg_merged[key[0] - 1,]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] - 2,]['MEAN(orientation)']) +
+                                                            abs(metrics_agg_merged[key[0] + 1,]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] + 2,]['MEAN(orientation)']) +
+                                                            abs(metrics_agg_merged[key[0] - 2,]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] - 3,]['MEAN(orientation)']) +
+                                                            abs(metrics_agg_merged[key[0] + 2,]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] + 3,]['MEAN(orientation)']))
+            except Exception as e:
+                # TODO: the warning below is raise mainly in cases of no SC segmentation, for example above the C1 level
+                #  --> consider processing only slices with SC segmentation
+                printv(f"WARNING: Failed to compute torsion for slice {key[0]}. Use `-v 2` flag to see details",
+                       verbose, type='warning')
+                if verbose == 2:
+                    printv(f"\t{e}", verbose)
+                metrics_agg_merged[key]['Torsion'] = None
+        else:
+            metrics_agg_merged[key]['Torsion'] = None
```

Comment:
One way that you can simplify this a little bit is to start with `= None` (as a "default" value), and this value will only be overwritten if the computation is successful:

```python
torsion = None
if key in slices:
    try:
       # [...]
    except:
       # [...]
metrics_agg_merged[key]['Torsion'] = torsion
```

Reply:
thanks 👍🏻 agree, done as part of https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4760/commits/b05b6f9883e98dc88aa4a49c8cbe1baf456bcd91